sycl/source/detail/pi_opencl.cpp

```diff
@@ -79,26 +76,21 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
-// TODO: implement portable call forwarding (ifunc is a GNU extension).
-// TODO: reuse same PI -> OCL mapping in pi_opencl.hpp, or maybe just
-//       wait until that one is completely removed.
-//
-#define _PI_CL(pi_api, ocl_api)             \
-static void *__resolve_##pi_api(void) {     \
-  return (void*) (ocl_api);                 \
-}                                           \
-decltype(ocl_api) OCL(pi_api) __attribute__((ifunc ("__resolve_" #pi_api)));
+// Forward calls to OpenCL RT.
+#define _PI_CL(pi_api, ocl_api)   \
+const decltype(::pi_api) * pi_api##OclPtr = \
```

Comment:
Another global variable, `pi_api`?

Reply:
This one is temporary. We could change the OpenCL PI plugin to be hooked like any other plugin (in run-time).